kitty graphics: performance enhancements

jcollie opened the pull request:
Improve the performance of Kitty graphics by switching to the WUFFS library for decoding PNG images and for "swizzling" G, GA, and RGB data to RGBA formats needed by the renderers.

WUFFS claims 2-3x performance benefits over other implementations, as well as memory-safe operations.

Although not thorougly benchmarked, performance is on par with Kitty's graphics decoding.

https://github.com/google/wuffs

This is a `perf` report of using `mpv` to play a video back using the Kitty image protocol from before the patch:

![before](https://github.com/user-attachments/assets/073b2de4-b02a-4087-ab6f-c43f33742b29)

After the patch:

![after](https://github.com/user-attachments/assets/4e317f3b-d9ca-423d-8dcc-7c215cc07f97)

jcollie committed:
kitty graphics: performance enhancements

Improve the performance of Kitty graphics by switching to the WUFFS
library for decoding PNG images and for "swizzling" G, GA, and RGB data
to RGBA formats needed by the renderers.

WUFFS claims 2-3x performance benefits over other implementations, as
well as memory-safe operations.

Although not thorougly benchmarked, performance is on par with Kitty's
graphics decoding.

https://github.com/google/wuffs

jcollie committed:
fix typos

mitchellh reviewed (changes requested):
Couple things I want to look at but a few big ones to request!

Thanks for doing this, the results look great.

jcollie committed:
kitty graphics: address review comments

- move wuffs code from src/ to pkg/
- eliminate stray debug logs
- make logs a warning instead of an error
- remove initialization of some structs to zero

jcollie committed:
remove unnecessary @constCast

mitchellh committed:
pkg/wuffs: define build defines separately, not in defs.zig

mitchellh committed:
some tweaks for wuffs

mitchellh committed:
renderer/metal: use wuffs for format conversion

mitchellh committed:
update zig cache hash

mitchellh committed:
pkg/wuffs: fix builds for 32-bit systems

mitchellh committed:
pkg/wuffs: use proper apple include paths on darwin targets
